src/native/http_request_utils.c

```diff
@@ -347,6 +372,36 @@ struct aws_http_message *aws_http_request_new_from_java_http_request(
     return NULL;
 }
 
+struct aws_http_headers *aws_http_headers_new_from_java_http_headers(JNIEnv *env, jbyteArray marshalled_headers) {
+    const char *exception_message = NULL;
+    struct aws_http_headers *headers = aws_http_headers_new(aws_jni_get_allocator());
+    if (headers == NULL) {
+        aws_jni_throw_runtime_exception(env, "aws_http_headers_new_from_java_http_headers: Unable to allocate request");
```

Comment:
Well, we don't fail from allocation anymore, but seems like it can still fail from arraylist init. Maybe just change the error message here.